src/ir_Sharp.cpp

```diff
@@ -339,7 +340,20 @@ void IRSharpAc::on(void) { setPower(true); }
 void IRSharpAc::off(void) { setPower(false); }
 
 void IRSharpAc::setPower(const bool on) {
-  setBit(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, on);
+  if (on)
+    setBits(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, 2, kSharpAcOnFromOff);
+  else
+    setBits(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, 2, kSharpAcOffFromOn);
```

Comment:
Ditto.
`src/ir_Sharp.cpp:346:  Lines should be <= 80 characters long  [whitespace/line_length] [2]`